File: library/src/main/java/com/mindera/skeletoid/generic/UIUtils.java
package com.mindera.skeletoid.generic;

import android.content.Context;
import android.support.annotation.VisibleForTesting;

public class UIUtils {

    @VisibleForTesting
    UIUtils() {
        throw new UnsupportedOperationException();
    }

    /**
     * Returns the status bar height in pixels.
     *
     * @param context       App context
     * @return status bar height or 0
     */
    public static int getStatusBarHeighPixels(Context context) {
        int resourceId = context.getResources()
                .getIdentifier("status_bar_height", "dimen", "android");

        return resourceId > 0 ? context.getResources().getDimensionPixelSize(resourceId) : 0;
    }
}

File: base/src/main/java/com/mindera/skeletoid/generic/UIUtils.kt
package com.mindera.skeletoid.generic

import android.content.Context

object UIUtils {

    /**
     * Returns the status bar height in pixels.
     *
     * @param context       App context
     * @return status bar height or 0
     */
    @JvmStatic
    fun getStatusBarHeightPixels(context: Context): Int {
        val resourceId = context.resources.getIdentifier("status_bar_height", "dimen", "android")
        return if (resourceId > 0) context.resources.getDimensionPixelSize(resourceId) else 0
    }

    /**
     * Given a value in dips, gets the amount of pixels that equates to in the given context.
     *
     * @param valueInDips - the amount of dips to convert to pixels
     * @param context - our current android context
     *
     * @return the value in pixels
     */
    fun getValueInPixels(valueInDips: Int, context: Context): Float {
        return (valueInDips * context.resources.displayMetrics.density)
    }
}